webapp/api/launchpad.py

```diff
@@ -1,6 +1,105 @@
-from launchpadlib.launchpad import Launchpad
+from hashlib import md5
+from webapp.api.requests import Session
 
 
-launchpad = Launchpad.login_anonymously(
-    "storefront", "production", version="devel"
-)
+class Launchpad:
+    """
+    Provides authentication for Launchpad.
+    """
+
+    BASE_URL = "https://api.launchpad.net"
+
+    def __init__(self, username, token, signature, session=Session()):
+        self.username = username
+        self.session = session
+        self.session.headers["Accept"] = "application/json"
+        self.session.headers["Authorization"] = (
+            f'OAuth oauth_version="1.0", '
+            f'oauth_signature_method="PLAINTEXT", '
+            f"oauth_consumer_key={username}, "
+            f'oauth_token="{token}", '
+            f'oauth_signature="&{signature}"'
+        )
+
+    def _request(self, method, url, params={}, data={}):
+        """
+        Makes a raw HTTP request and returns the response.
+        """
+        if not url.startswith("http"):
```

Comment:
When are we using a URL that doesn't start with `BASE_URL`? It would be much neater to keep this class *explicitly* to calls made to the same API...

Reply:
You are right. The response of a snap return properties with links and I was just using this to call those URLs.